src/app/util/attribute-storage.h

```diff
@@ -241,15 +241,35 @@ bool emberAfEndpointIsEnabled(chip::EndpointId endpoint);
 uint8_t emberAfGetClusterCountForEndpoint(chip::EndpointId endpoint);
 const EmberAfCluster * emberAfGetClusterByIndex(chip::EndpointId endpoint, uint8_t clusterIndex);
 
-uint16_t emberAfGetDeviceIdForEndpoint(chip::EndpointId endpoint);
+//
+// Retrieve the device type list associated with a specific endpoint.
+//
+const chip::Span<const EmberAfDeviceType> emberAfDeviceTypeListFromEndpoint(chip::EndpointId endpoint);
+
+//
+// Over-ride the device type list current associated with an endpoint with a user-provided list.
+//
+// NOTE: It is the application's responsibility to free the existing list that is being replaced if needed.
```

Comment:
Should also document that the span needs to point to a buffer that will live long enough.